File: tests/fixtures/step-defs/python/bad/failing_steps.py
"""Bad step definitions - should fail quality checks."""
from pytest_bdd import given, when, then


@given("a user exists")
def given_user_exists():
    pass


@when("the user performs an action")
def when_user_acts():
    pass


@then("the result is correct")
def then_result_correct():
    assert True


@then("everything works")
def then_everything_works():
    assert 1


@then("the data is saved")
def then_data_saved():
    """This then step has no assertion at all."""
    print("checking data...")
    x = 1 + 1

File: tests/fixtures/step-defs/javascript/bad/failing.steps.js
const { Given, When, Then } = require('@cucumber/cucumber');
const { expect } = require('chai');

Given('a user exists', function () {
});

When('the user performs an action', function () {
});

Then('the result is correct', function () {
    expect(true).toBe(true);
});

Then('the data is saved', function () {
    console.log('checking data...');
    const x = 1 + 1;
});
